### PR TITLE
Fix `test_vec_normalize.py`, `test_tensorboard.py` and `common/monitor.py` type hint

### PR DESCRIPTION
**Pull request by qgallouedec**
## Description

In this PR, I also replaced `Union[int, float]` by just `float`, as they are equivalent for type hint: 
 In https://peps.python.org/pep-0484/#the-numeric-tower
> Rather than requiring that users write import numbers and then use numbers.Float etc., this PEP proposes a straightforward shortcut that is almost as effective: when an argument is annotated as having type float, an argument of type int is acceptable

> **Warning**
> This is only true for type hint, and you can replace `isinstance(x, (int, float))` by `isinstance(x, float)`

## Motivation and Context

- [ ] I have raised an issue to propose this change ([required](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) for new features and bug fixes)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist:

- [ ] I've read the [CONTRIBUTION](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) guide (**required**)
- [ ] I have updated the changelog accordingly (**required**).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.
- [ ] I have reformatted the code using `make format` (**required**)
- [ ] I have checked the codestyle using `make check-codestyle` and `make lint` (**required**)
- [ ] I have ensured `make pytest` and `make type` both pass. (**required**)
- [ ] I have checked that the documentation builds using `make doc` (**required**)

Note: You can run most of the checks using `make commit-checks`.

Note: we are using a maximum length of 127 characters per line

### CODE REVIEW

File: tests/test_tensorboard.py [hunk 1]
```diff
@@ -18,21 +21,19 @@
...
-        hparam_dict = {
-            "algorithm": self.model.__class__.__name__,
-            "learning rate": self.model.learning_rate,
-            "gamma": self.model.gamma,
-        }
+        hparam_dict: Dict[str, Union[str, float]] = {"algorithm": self.model.__class__.__name__}
+        if isinstance(self.model.learning_rate, float):  # Can also be Schedule, in that case, we don't report
+            hparam_dict["learning rate"] = self.model.learning_rate
+        if isinstance(self.model, (OnPolicyAlgorithm, OffPolicyAlgorithm)):
+            hparam_dict["gamma"] = self.model.gamma
```
Comment (qgallouedec) [inline tests/test_tensorboard.py:33]: Currently, gamma is defined in `OnPolicyAlgorithm` and `OffPolicyAlgorithm`, but not in `BaseAlgorithm`. I am not very satisfied with this solution, but it works.

Comment (araffin) [reply tests/test_tensorboard.py:33]: gamma makes sense for most algorithms but not ES ones like ARS/CEM (actually we could set gamma=1 for them for consistency even though it is not used)

Comment (araffin) [reply tests/test_tensorboard.py:33]: > I am not very satisfied with this solution, but it works.

yes, not pretty but:
1. it is for a test
2. you could probably ignore mypy here

**Review by araffin — COMMENTED** [review]
What was missing for that one?

**Comment by qgallouedec** [comment]
The CI didn't pass, so I put it off. But if the CI passes now, I think it 's ready to be merged.

**Review by araffin — APPROVED** [review]

**Pull request merged**